### ISSUE TITLE
Is the maxpool and avgpool interchanged here?

### ISSUE THREAD

**Issue by Noplz** [Cadene/pretrained-models.pytorch#15]
https://github.com/Cadene/pretrained-models.pytorch/blob/dbacab66ca044c05eafd5978a14e85947a53e258/pretrainedmodels/nasnet.py#L82-L86

I compared the tensorflow implementation, I think your implementation in ReductionCell0 is correct and the CellStem0 should stay the same as ReductionCell0.

**Comment by shicai** [comment]
@Noplz I think you are right. it should be max pooling for `comb_iter_1_left ` and average pooling for `comb_iter_2_left `.

**Comment by aussetg** [comment]
I also don't understand why there are Cell1 and Cell0, the original paper only includes one type of Normal Cell, and one type of Reduction Cell

**Comment by Cadene** [comment]
@Noplz @shicai  You were right, I made a mistake. I fixed it in this commit https://github.com/Cadene/pretrained-models.pytorch/commit/40f4f7eb31bcdb08cbc5d178ea5653563e0ea561

@aussetg An more refactored version is available in Tensorflow https://github.com/taehoonlee/tensornets/blob/master/tensornets/nasnets.py
However, it was easier for me to create CellStem0, CellStem1, Cell0, Cell1 and ReductionCell0 regarding my will to port the pretrained parameters from tensorflow after my deadlin (15 nov).

**Comment by aussetg** [comment]
@Cadene How do you port weights from TensorFlow?

Also I don't think the TF weights are sota, they don't implement path wise dropout like in the paper ( probably because they can't with TensorFlow as the graph then changes at every iteration. But then I wonder what framework the Google team used 😏 )

**Comment by Noplz** [comment]
@Cadene great job anyway, this repo is really helpful :) 👍

**Comment by rkaplan** [comment]
Congrats on finishing your deadline :) just wanted to say thank you for working on this NASNet port, looking forward to trying it out!

**Comment by Cadene** [comment]
@rkaplan @Noplz @aussetg Done ! https://github.com/Cadene/pretrained-models.pytorch/commit/247b0375abb6120ab4ef5642be59aee9d9e5542d
